### PR TITLE
Enable `gci` linter for test Packages

### LINKED ISSUE #2962: Enable linters for test packages
**Is your feature request related to a problem? Please describe.**
As of now linters are disabled from test packages in .golangci.yaml file:
```
issues:
  exclude-rules:
    - path: '_test.go'
      linters:
        - errcheck                      # Errors may be ignored in tests.
        - unparam                       # Tests might have unused function parameters.
        - lll
        - dupl
        - misspell
        - nestif
        - gci
```
**Describe the solution you'd like**
After enabling this linter, fix all kind of linter issues coming up in test packages.

### PR DESCRIPTION
**Pull request by anishbista60**
## Change Overview

Enable `gci` linter for test Packages
## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test
- [ ] :building_construction: Build

## Issues 

- fixes #2962

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Comment by anishbista60** [comment]
@julio-lopez Work fine in my machine . Please test it .

**Comment by anishbista60** [comment]
@viveksinghggits here also. Please test it . It works fine in my machine.

**Review by julio-lopez — CHANGES_REQUESTED** [review]

File: pkg/apis/cr/v1alpha1/repositoryserver_types_test.go
Comment (anishbista60) [reply pkg/apis/cr/v1alpha1/repositoryserver_types_test.go:24]: @julio-lopez i have arranged packages based on this rule .
```
  gci:
    sections:
      - standard
      - default
      - prefix(github.com/kanisterio/kanister)
      - blank
      - dot
```
sure , let's wait for another review from @viveksinghggits

Comment (viveksinghggits) [reply pkg/apis/cr/v1alpha1/repositoryserver_types_test.go:24]: In some of the PRs we have treated `_` blank imports as separate category https://github.com/kanisterio/kanister/pull/2500#discussion_r1675602433 and that's the reason I considered that an acceptable change.

@anishbista60, I am assuming that gci is recommending `. "gopkg.in/check.v1"` to be grouped separately just because it's a dot import. 

Now, if we plan to remove the `.` and make it a standard import, we can easily group it with the third party imports but if it's being imported as `.` or `_` I think gci is going to recommend it to group separately.

Comment (viveksinghggits) [reply pkg/apis/cr/v1alpha1/repositoryserver_types_test.go:24]: To make things clear I am not recommending that we should change the `.` import to standard import in this PR itself, we can do that later as well. But if we have `.` or `_` blank import, gci would recommend to group that separately.

Comment (julio-lopez) [reply pkg/apis/cr/v1alpha1/repositoryserver_types_test.go:24]: The _"right thing to do"_ here would be to remove the `.` import for `check.v1` and replace all the call sites to include the `check` prefix.

Comment (viveksinghggits) [reply pkg/apis/cr/v1alpha1/repositoryserver_types_test.go:24]: > The _"right thing to do"_ here would be to remove the `.` import for `check.v1` and replace all the call sites to include the `check` prefix.

I agree. Should we do that as part of this PR itself.

Comment (julio-lopez) [reply pkg/apis/cr/v1alpha1/repositoryserver_types_test.go:24]: > I agree. Should we do that as part of this PR itself?

- Right now is as good of a time as ever.
- Those changes should be done before enabling the `gci` linter, otherwise this has the risk of falling through the cracks and never getting done.

Comment (viveksinghggits) [reply pkg/apis/cr/v1alpha1/repositoryserver_types_test.go:24]: Sounds good. @anishbista60 let's make that change. Please let me know if you have some questions about how to implement this.

Comment (viveksinghggits) [reply pkg/apis/cr/v1alpha1/repositoryserver_types_test.go:24]: yes, that looks like the correct order. also, can you please check how `goimports` handles the blank and `.` imports.

Comment (viveksinghggits) [reply pkg/apis/cr/v1alpha1/repositoryserver_types_test.go:24]: does it treat them as separate group or not?

Comment (anishbista60) [reply pkg/apis/cr/v1alpha1/repositoryserver_types_test.go:24]: @viveksinghggits goimports does not treat blank (_) and dot (.) imports as a separate group by default.

Comment (viveksinghggits) [reply pkg/apis/cr/v1alpha1/repositoryserver_types_test.go:24]: ok.

Comment (anishbista60) [reply pkg/apis/cr/v1alpha1/repositoryserver_types_test.go:24]: @viveksinghggits There is are so many places that i am facing error after removing `.`  Should i replace it with `check` every where ?

for example : 

- `c.Assert(err, check.IsNil)` is throwing error . So, it change it to `c.Assert(err, check.IsNil)`
- `c.Assert(snaps,Not(HasLen),0)`   to  `c.Assert(snaps, check.Not(check.HasLen), 0)`

Comment (anishbista60) [reply pkg/apis/cr/v1alpha1/repositoryserver_types_test.go:24]: Let me make changes to some file and push it . After reviewing , we can move futher.

**Comment by anishbista60** [comment]
@julio-lopez now , let's merge this .

**Review by julio-lopez — APPROVED** [review]

**Pull request merged**
